webapp/channels/src/components/more_channels/more_channels.tsx

```diff
@@ -96,9 +99,20 @@ export default class MoreChannels extends React.PureComponent<Props, State> {
         if (this.props.canShowArchivedChannels) {
             this.props.actions.getArchivedChannels(this.props.teamId, 0, CHANNELS_CHUNK_SIZE * 2);
         }
+        if (this.props.channels.length > 0 || this.props.archivedChannels.length > 0) {
```

Comment:
Do we need to call this on mount? I'd think we should wait until we've received the channels until we make this request. Perhaps we should have a `.then` on the initial call to `getChannels`/`getArchivedChannels` and then on the call to `getChannels` in `getNextPage`?

Reply:
Good catch especially on the `nextPage`. My idea was to combine `channels` and `archivedChannels` and make a single call to backend. However if user doesnt select to see archived channels, it would be a redundant call to the back end. I updated as suggested please let me know if I am missing anything @hmhealey

Reply:
good point. What you are suggesting is the best way to handle this in a single call. updated as suggested